### PR TITLE
Gh4662 safe AD_element in tabs, window and menus

### PR DESCRIPTION
**Pull request by metas-rc**
* a safer version that supports this functionality on different customer systems that are not identical with what we have on metasfresh master

https://github.com/metasfresh/metasfresh/issues/4662

### CODE REVIEW

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MWindow.java [hunk 1]
```diff
@@ -53,7 +54,7 @@ public MWindow (Properties ctx, int AD_Window_ID, String trxName)
 		super (ctx, AD_Window_ID, trxName);
 		if (AD_Window_ID == 0)
 		{
-			setWindowType (WINDOWTYPE_Maintain);	// M
+			setWindowType (WINDOWTYPE_Verwalten);	// M
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MWindow.java]: pls use English names

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/translation/api/IElementTranslationBL.java [hunk 2]
```diff
@@ -48,9 +48,21 @@
 
 	/**
 	 * Update Fields translation for the fields that have the element given as parameter as AD_Name_ID, delete translation if AD_Name was deleted
-	 * 
+	 *
 	 * @param ad_Element_ID
 	 */
 	void updateFieldTranslationsFromAD_Name(int ad_Element_ID);
 
+	void updateWindowTranslationsFromElement(int elementId);
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java/de/metas/translation/api/IElementTranslationBL.java]: always use object IDs when possible/available.
i.e. in this case, use AdElementId

File: de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/menu/model/interceptor/AD_Menu.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	public void init()
+	{
+		Services.get(IProgramaticCalloutProvider.class).registerAnnotatedCallout(this);
+	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = I_AD_Tab.COLUMNNAME_AD_Element_ID)
+	@CalloutMethod(columnNames = I_AD_Menu.COLUMNNAME_AD_Element_ID)
+	public void onElementIDChanged(final I_AD_Menu menu) throws SQLException
+	{
+		if (!IADMenuDAO.DYNATTR_AD_Menu_UpdateTranslations.getValue(menu, true))
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/menu/model/interceptor/AD_Menu.java]: that DYNATTR_ shall go to one of the Element*BL classes.

**Comment by metas-ts** [comment]
hi @metas-rc what is the relation between this PR and the 3 that i just assigned to you (mostly because they don't build), i.e.:
* https://github.com/metasfresh/metasfresh/pull/4731
* https://github.com/metasfresh/mo73/pull/501
* https://github.com/metasfresh/metasfresh/pull/4670
?

**Comment by metas-rc** [comment]
Hi

This PR is for the task #4662. I agreed with Teo to not do the fixes here because they were already fixed on gh_4663_safe. So please, ignore this PR, the one from #4663 also contains everything from #4662 so they can be merged together.


https://github.com/metasfresh/metasfresh/pull/4670 can be ignored ( closed) . I only brought some commits from it and put them in the new branch, gh4663_safe

For mo73 I did a new PR now with the branch gh4663_safe . I forgot to do it before.

https://github.com/metasfresh/metasfresh/pull/4731 is rebuilding after I updated from master. Let's see if it works now.

**Comment by metas-rc** [comment]
Please, review together with https://github.com/metasfresh/metasfresh/issues/4663

**Pull request closed without merging**